packages/api/src/workerCreateIssue.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  MeetupFormValues,
+  getMeetupIssueBody,
+  meetupFormValuesSchema,
+} from 'meetup-shared';
+import { App } from 'octokit';
+import { safeParseAsync } from 'valibot';
+import { Env } from './workerEnv';
+
+export async function parseCreateIssueReqBody(req: Request): Promise<
+  | {
+      errorResponse: Response;
+      parsedMeetup?: never;
+    }
+  | {
+      errorResponse?: never;
+      parsedMeetup: MeetupFormValues;
+    }
+> {
+  try {
+    const json = await req.json();
+
+    const meetupFormValues = await safeParseAsync(meetupFormValuesSchema, json);
+
+    if (!meetupFormValues.success) {
+      const issues = meetupFormValues.error.issues;
+
+      console.error('Invalid JSON - validation error - ', issues);
+
+      return {
+        errorResponse: new Response(
+          JSON.stringify({
+            error: {
+              message: 'Invalid JSON - validation error',
+              details: issues,
+            },
+          }),
+          {
+            status: 400,
+            headers: { 'Content-Type': 'application/json' },
+          },
+        ),
+      };
+    }
+
+    return { parsedMeetup: meetupFormValues.data };
+  } catch (e) {
+    console.error('Invalid JSON - catched an error - ', e);
+
+    return {
+      errorResponse: new Response(
+        JSON.stringify({
+          error: {
+            message: 'Invalid JSON - catched an error',
+            details: e,
+          },
+        }),
+        {
+          status: 400,
+          headers: { 'Content-Type': 'application/json' },
+        },
+      ),
+    };
+  }
+}
+
+export async function createIssue(props: {
+  meetupFormValues: MeetupFormValues;
+  env: Env;
+}): Promise<
+  | {
+      errorResponse: Response;
+      data?: never;
+    }
+  | {
+      errorResponse?: never;
+      data: {
+        issueUrl: string;
+        issueNumber: number;
+      };
```

Comment:
Tässä ja parissa muussa vastaavassa conditional typessä voidaan suorilta jättää `never` kentät poissa. Tällä tavalla ei tarvitse duplikoida kenttiä ja ylläpitää kahdessa kohtaa kun uutta lisätään. Eli:

```suggestion
  | {
      errorResponse: Response;
    }
  | {
      data: {
        issueUrl: string;
        issueNumber: number;
      };
```

Ja sitten näiden kutsuja tekee:

```ts
const response = await createIssue(...);

if("errorResponse" in response) {
   // Tässä kohtaa tyypitys on { errorResponse }
   // ja response.data ei ole olemassa
   return;
}

// Tässä kohtaa tyypitys on { data: ... }
```

Tästähän saisi jonkin generic helperin jopa:
```ts
type ApiResponse<Data, Error> = { data: Data } | { error: Error }
```

Reply:
Tää on kova! 👍🏻 En tiennykkää, että tuo `'prop' in ...` syntaksi hoitaa tyypit tolla tavalla. Pitäskö tehä pelkästään workkerissa käytettäväksi tällänen helpperi?:

```ts
type DataOrErr<Data> = { data: Data } | { errorResponse: Response }
```